keps/sig-node/2400-node-swap/README.md

```diff
@@ -0,0 +1,847 @@
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubernetes currently does not support the use of [swap
+memory](https://en.wikipedia.org/wiki/Paging#Linux) on Linux, as it is
+difficult to provide guarantees and account for pod memory utilization when
+swap is involved. As part of Kubernetes’ earlier design, [swap support was
+considered out of scope](https://github.com/kubernetes/kubernetes/issues/7294).
+
+However, there are a [number of use cases](#user-stories) that would benefit
+from Kubernetes nodes supporting swap. Hence, this proposal aims to add swap
+support to nodes in a controlled, predictable manner so that Kubernetes users
+can perform testing and provide data to continue building cluster capabilities
+on top of swap.
+
+## Motivation
+
+There are two distinct types of user for swap, who may overlap:
+- node administrators, who may want swap available for node-level performance
+  tuning and stability/reducing noisy neighbour issues
+- application developers, who have written applications that would benefit from
+  using swap memory
+
+There are hence a number of possible ways that one could envision swap use on a
+node.
+
+### Scenarios
+
+1. Swap is enabled on a node's host system, but the kubelet does not permit
+   Kubernetes workloads to use swap. (This scenario is a prerequisite for the
+   following use cases.)
+1. Swap is enabled at the node level. The CRI can be globally configured to
```

Comment:
One clarification is that the kubelet will still pass down the exact swap setting over the CRI per container. The setting itself will be for the kubelet but CRI doesn't have a global setting as such.